x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_embeddable.tsx

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Suspense } from 'react';
+import ReactDOM from 'react-dom';
+import { pick } from 'lodash';
+
+import { Embeddable } from '@kbn/embeddable-plugin/public';
+
+import { CoreStart } from '@kbn/core/public';
+import { i18n } from '@kbn/i18n';
+import { Subject } from 'rxjs';
+import { KibanaContextProvider, KibanaThemeProvider } from '@kbn/kibana-react-plugin/public';
+import { IContainer } from '@kbn/embeddable-plugin/public';
+import { DatePickerContextProvider, type DatePickerDependencies } from '@kbn/ml-date-picker';
+import { UI_SETTINGS } from '@kbn/data-plugin/common';
+import { EmbeddableSingleMetricViewerContainer } from './embeddable_single_metric_viewer_container_lazy';
+import type { JobId } from '../../../common/types/anomaly_detection_jobs';
+import type { MlDependencies } from '../../application/app';
+import {
+  ANOMALY_EXPLORER_CHARTS_EMBEDDABLE_TYPE,
+  SingleMetricViewerEmbeddableInput,
+  AnomalyChartsEmbeddableOutput,
+  SingleMetricViewerServices,
+} from '..';
+import { EmbeddableLoading } from '../common/components/embeddable_loading_fallback';
+
+export const getDefaultSingleMetricViewerPanelTitle = (jobIds: JobId[]) =>
+  i18n.translate('xpack.ml.anomalyChartsEmbeddable.title', {
+    defaultMessage: 'ML single metric viewer chart for {jobIds}',
+    values: { jobIds: jobIds.join(', ') },
+  });
+
+export type ISingleMetricViewerEmbeddable = typeof SingleMetricViewerEmbeddable;
+
+export class SingleMetricViewerEmbeddable extends Embeddable<
+  SingleMetricViewerEmbeddableInput,
+  AnomalyChartsEmbeddableOutput
+> {
+  private node?: HTMLElement;
+  private reload$ = new Subject<void>();
+  public readonly type: string = ANOMALY_EXPLORER_CHARTS_EMBEDDABLE_TYPE;
+
+  constructor(
+    initialInput: SingleMetricViewerEmbeddableInput,
+    public services: [CoreStart, MlDependencies, SingleMetricViewerServices],
+    parent?: IContainer
+  ) {
+    super(initialInput, {} as AnomalyChartsEmbeddableOutput, parent);
+  }
+
+  public onLoading() {
+    this.renderComplete.dispatchInProgress();
+    this.updateOutput({ loading: true, error: undefined });
+  }
+
+  public onError(error: Error) {
+    this.renderComplete.dispatchError();
+    this.updateOutput({ loading: false, error: { name: error.name, message: error.message } });
+  }
+
+  public onRenderComplete() {
+    this.renderComplete.dispatchComplete();
+    this.updateOutput({ loading: false, error: undefined });
+  }
+
+  public render(node: HTMLElement) {
```

Comment:
The embeddable doesn't seem to be using the custom time range panel settings. It is always using the full dashboard time range.

<img width="303" alt="image" src="https://github.com/elastic/kibana/assets/7405507/09a4b16b-e6a2-4aa1-9808-dbcdd221a691">

Reply:
Fixed in d931543fa904ca44d10ca6e19a0855dab00062d6